### PR TITLE
Escaping special regex chars

### PR DESCRIPTION
**Pull request by ReinierCC**
To properly sanitize input strings we have to escape any special characters.

Wanted to get some feedback on this implementation, where I manually specified the special characters and are escaping.

Some other potential routes would be to use a library with this feature but, this would create a new dependency.

Currently have a simple test function in the same file. Can move this to a more appropriate location after feedback.

### CODE REVIEW

File: src/panels/TcpDumpPanel.ts [hunk 1]
```diff
@@ -24,7 +24,14 @@ const tcpDumpCommandBase = "tcpdump --snapshot-length=0 -vvv";
 const captureDir = "/tmp";
 const captureFilePrefix = "vscodenodecap_";
 const captureFileBasePath = `${captureDir}/${captureFilePrefix}`;
-const captureFilePathRegex = `${captureFileBasePath.replace(/\//g, "\\$&")}(.*)\\.cap`; // Matches the part of the filename after the prefix
+const captureFileBasePathEscaped = escapeRegExp(captureFileBasePath);
+const captureFilePathRegex = `${captureFileBasePathEscaped}(.*)\\.cap`; // Matches the part of the filename after the prefix
+
+// Escape all regex meta characters to ensure sanitation and '/' for later use in regex pattern
+function escapeRegExp(input: string): string {
```
Comment (Tatsinnit) [inline src/panels/TcpDumpPanel.ts:31]: Thanks for this. 👍 worth to just check what is the standard `tcpDump` file name structure - I believe anything standardised file naming convention + `*.pcap` but worth to just verify please.

File: src/panels/TcpDumpPanel.ts [hunk 2]
```diff
@@ -53,6 +60,20 @@ function getCaptureFromFilePath(filePath: string): string | null {
     return fileMatch && fileMatch[1];
 }
 
+// Test function to verify escaping
+function testEscapeRegExp() {
```
Comment (Tatsinnit) [inline src/panels/TcpDumpPanel.ts:64]: 💡 Idea: Can we please move this to the util file for latter wider consumption? if you think its a good idea? we can then test it via test suite run.

File: src/panels/TcpDumpPanel.ts (hunk 1 above)
Comment (ReinierCC) [reply src/panels/TcpDumpPanel.ts]: Yes! you're right. ".pcap" is the standard convention. Currently ".cap" is referenced in many places so if we want to change the convention would probably be best for a separate PR. Lmn what you think.

File: src/panels/TcpDumpPanel.ts (hunk 2 above)
Comment (ReinierCC) [reply src/panels/TcpDumpPanel.ts]: Added a mocha test suite for the escaping function

**Review by hsubramanianaks — APPROVED** [review]
lgtm, just make sure to run a manual test for tcp dump to make sure it works. Thanks for finding this and fixing it @ReinierCC

**Pull request merged**